src/main/java/mate/controller/car/AddCarController.java

```diff
@@ -0,0 +1,42 @@
+package mate.controller.car;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Driver;
+import mate.service.CarService;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/cars/add")
+public class AddCarController extends HttpServlet {
+    private final Injector injector = Injector.getInstance("mate");
+    private CarService carService = (CarService) injector.getInstance(CarService.class);
+    private final ManufacturerService manufacturerService =
+            (ManufacturerService) injector.getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.setAttribute("manufacturersList", manufacturerService.getAll());
+        req.getRequestDispatcher("/WEB-INF/views/car/addCar.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String model = req.getParameter("model");
+        Long manufacturerId = Long.valueOf(req.getParameter("manufacturer_id"));
+        Car car = new Car();
+        car.setManufacturer(manufacturerService.get(manufacturerId));
+        car.setModel(model);
+        car.setDrivers(new ArrayList<Driver>());
+        carService.create(car);
+        resp.sendRedirect(req.getContextPath() + "/cars/add");
```

Comment:
After creating a new car, it's better to redirect the user to the list of cars instead of the add car page. Change the redirect URL to '/cars/all'.